bundle/edu.gemini.phase2.skeleton.servlet/src/main/scala/edu/gemini/phase2/skeleton/factory/SpProgramFactory.scala

```diff
@@ -179,47 +179,42 @@ object SpProgramFactory {
     itacAcceptance(proposal).exists(_.rollover)
 
   def isThesis(proposal: Proposal): Boolean =
-    proposal.investigators.all exists { i => i.status == GRAD_THESIS }
+    proposal.investigators.all.exists(_.status == GRAD_THESIS)
 
   def gemEmail(proposal: Proposal): Option[String] =
     itacAcceptance(proposal).flatMap(_.contact)
 
   def minBand3Time(proposal: Proposal): Option[TimeValue] =
     proposal.proposalClass match {
-      case q: QueueProposalClass => q.band3request map { r => toTimeValue(r.minTime) }
+      case q: QueueProposalClass => q.band3request.map(r => toTimeValue(r.minTime, proposal.programTimeRatio))
       case _                     => None
     }
 
-  private def toTimeValue(ta: TimeAmount): TimeValue =
+  private def toTimeValue(ta: TimeAmount, ratio: Double = 1.0): TimeValue =
     ta.units match {
-      case TimeUnit.NIGHT => new TimeValue(ta.value, TimeValue.Units.nights)
-      case _              => new TimeValue(ta.toHours.value, TimeValue.Units.hours)
+      case TimeUnit.NIGHT => new TimeValue(ta.value * ratio, TimeValue.Units.nights)
+      case _              => new TimeValue(ta.toHours.value * ratio, TimeValue.Units.hours)
     }
 
   def timeAcctAllocation(proposal: Proposal): Option[TimeAcctAllocation] =
-    awardedHours(proposal).filter(_ > 0.0) flatMap { hrs =>
-      val progTime = hoursFromObservations(proposal, _.progTime)
-      val partTime = hoursFromObservations(proposal, _.partTime)
-
+    // We ignore the awarded hours here and work instead with the program time and partner time.
+    awardedHours(proposal).filter(_ > 0.0) flatMap { _ =>
       timeAccountingRatios(proposal) match {
         case Nil => None
         case ratios =>
           val jmap = ratios.map { case (cat, rat) =>
             def durationRatio(v: TimeAmount): Duration =
               Duration.ofMillis(((v.hours * rat) * MsPerHour).round)
 
-            val award = new TimeAcctAward(durationRatio(progTime), durationRatio(partTime))
+            val award = new TimeAcctAward(durationRatio(proposal.programTime), durationRatio(proposal.partnerTime))
```

Comment:
Ooops. I saw this after I merged. I'm fine either way, as the values should be the same, but I'll change this.